src/components/frame/Extensions/pages/ExtensionDetailsPage/partials/ExtensionSettings/utils.tsx

```diff
@@ -0,0 +1,227 @@
+import { Extension } from '@colony/colony-js';
+import React from 'react';
+import { type FieldValues } from 'react-hook-form';
+import { toast } from 'react-toastify';
+
+import { type RefetchColonyFn } from '~context/ColonyContext/ColonyContext.ts';
+import { ExtensionDetailsPageTabId } from '~frame/Extensions/pages/ExtensionDetailsPage/types.ts';
+import {
+  waitForExtensionPermissions,
+  waitForDbAfterExtensionAction,
+} from '~frame/Extensions/pages/ExtensionDetailsPage/utils.tsx';
+import {
+  ExtensionMethods,
+  type RefetchExtensionDataFn,
+} from '~hooks/useExtensionData.ts';
+import { ActionTypes } from '~redux';
+import Toast from '~shared/Extensions/Toast/index.ts';
+import { type OnSuccess } from '~shared/Fields/index.ts';
+import { type SetStateFn } from '~types';
+import {
+  type ExtensionInitParam,
+  type AnyExtensionData,
+} from '~types/extensions.ts';
+import {
+  convertFractionToEth,
+  isInstalledExtensionData,
+} from '~utils/extensions.ts';
+
+import {
+  type MultiSigSettingsFormValues,
+  MultiSigThresholdType,
+} from './MultiSigSettings/types.ts';
+import {
+  getDomainThresholds,
+  getGlobalThresholdType,
+} from './MultiSigSettings/utils.ts';
+
+export const getFormSuccessFn =
+  <T extends FieldValues>({
+    setWaitingForActionConfirmation,
+    extensionData,
+    refetchColony,
+    refetchExtensionData,
+    setActiveTab,
+  }: {
+    setWaitingForActionConfirmation: SetStateFn;
+    setActiveTab: (tabId: ExtensionDetailsPageTabId) => void;
+    extensionData: AnyExtensionData;
+    refetchColony: RefetchColonyFn;
+    refetchExtensionData: RefetchExtensionDataFn;
+  }): OnSuccess<T> =>
+  async (fieldValues, { reset }) => {
+    setWaitingForActionConfirmation(true);
+    const isSaveChanges =
+      extensionData &&
+      isInstalledExtensionData(extensionData) &&
+      (extensionData.isInitialized || extensionData.configurable);
+
+    try {
+      /* Wait for permissions first, so that the permissions warning doesn't flash in the ui */
+      await waitForExtensionPermissions({ extensionData, refetchColony });
```

Comment:
This still showed up for me when installing the Staking extension

Reply:
@rdig Are you positive it was when installing Staked Expenditure and not Staged? I can't replicate it for Staked, while it is expected to show for Staged extension